src/utils/AutoControlledComponent-slim.js

```diff
@@ -0,0 +1,44 @@
+/* eslint-disable */
+import _ from 'lodash'
+import { Component } from 'react'
+
+const toDefault = (prop) => `default${prop[0].toUpperCase() + prop.slice(1)}`
+
+export default class AutoControlledComponent extends Component {
+  componentWillMount() {
+    const { autoControlledProps, defaultProps, name, propTypes } = this.constructor
+
+    this.state = _.transform(autoControlledProps, (res, prop) => {
+      const defaultProp = toDefault(prop)
+
+      // apply default props if they exist
+      res[prop] = this.props[defaultProp in this.props ? defaultProp : prop]
+    }, {})
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (super.componentWillReceiveProps) super.componentWillReceiveProps(nextProps)
```

Comment:
Hm, this seems intuitive.  However, when removing the checks we get this in tests:

```shell
undefined is not an object (evaluating '_get(Object.getPrototypeOf(AutoControlledComponent.prototype), 'componentWillReceiveProps', this).call')
```

I'm not sure how React is constructing these classes, but apparently they do not have all their methods at least in tests.